src/presentation/ui/page/mappings.templ

```diff
@@ -0,0 +1,338 @@
+			@componentForm.SubmitButton("Create", "ph-check-fat", "closeCreateVhostModal()", false, true)
+		</template>
+		<template x-if="virtualHost.hostname != ''">
+			@componentForm.SubmitButton("Create", "ph-check-fat", "closeCreateVhostModal()", false, false)
+		</template>
+	</form>
+}
+
+templ CreateVhostModal() {
+	@componentStructural.Modal(
+		"Create Virtual Host", "isCreateVhostModalOpen", "closeCreateVhostModal()",
+		"",
+	) {
+		@VirtualHostRegularForm()
+	}
+}
+
+func filterOnlyVhostsHostnames(vhostsWithMappings []dto.VirtualHostWithMappings) []string {
+	vhostsHostnames := []string{}
+	for _, vhostWithMapping := range vhostsWithMappings {
+		vhostsHostnames = append(vhostsHostnames, vhostWithMapping.Hostname.String())
+	}
+
+	return vhostsHostnames
+}
+
+func getOnlyServiceNames() []string {
+	onlyServiceNames := []string{}
+	for serviceName := range valueObject.NativeSvcNamesWithAliases {
+		onlyServiceNames = append(onlyServiceNames, serviceName)
+	}
+
+	return onlyServiceNames
+}
+
+templ MappingRegularForm(vhostsWithMappings []dto.VirtualHostWithMappings) {
+	<form
+		hx-post="/api/v1/vhosts/mapping/"
+		hx-indicator="#loading-overlay"
+		hx-swap="none"
+	>
+		<div class="-mt-4 my-6 space-y-6">
+			@componentMappings.MappingTypeSelectInput("targetType", "Type", "mapping.targetType", "clearMappingTargetValue()")
+			@componentMappings.MappingTypeInfoDialog()
+			<template x-if="isCreateMappingFromVhost">
+				@componentForm.ReadOnlyField("text", "hostname", "Hostname", "virtualHost.hostname", false)
+			</template>
+			<template x-if="!isCreateMappingFromVhost">
+				@componentForm.SelectInput("hostname", "Hostname", "virtualHost.hostname", filterOnlyVhostsHostnames(vhostsWithMappings), true, false)
+			</template>
+			@componentForm.InputFieldWithPrefix("text", "path", "Source URL", "mapping.path", "vhostWithTrailingSlash", false)
+		</div>
+		<template x-if="mapping.targetType == 'service'">
+			<div class="mb-6 space-y-6">
+				@componentForm.SelectInput("targetValue", "Service Name", "mapping.targetValue", getOnlyServiceNames(), true, false)
+			</div>
+		</template>
+		<template x-if="mapping.targetType == 'url'">
+			<div class="mb-6 space-y-6">
+				@componentForm.InputField("text", "targetValue", "Destination URL", "mapping.targetValue", false)
+			</div>
+		</template>
+		<template x-if="mapping.targetType == 'inline-html'">
+			<div class="mb-6 space-y-6">
+				@componentForm.TextArea("targetValue", "Inline HTML", "mapping.targetValue", false)
+				@componentMappings.ResponseCodeSelectInput("targetHttpResponseCode", "Response Code", "mapping.targetHttpResponseCode")
+			</div>
+		</template>
+		<template x-if="mapping.targetType == 'response-code'">
+			<div class="mb-6 space-y-6">
+				@componentMappings.ResponseCodeSelectInput("targetValue", "Response Code", "mapping.targetValue")
+				<input type="hidden" x-model="mapping.targetValue" x-bind:value="mapping.targetHttpResponseCode"/>
```

Comment:
This is weird. Why are you hidden a input type which model is targetValue but it's binded to response code?

Reply:
Useless. Removed!